flight/Libraries/math/pid.c

```diff
@@ -199,6 +230,67 @@ void pid_configure(struct pid *pid, float p, float i, float d, float iLim)
 }
 
 /**
+ * Assigns a pid_deadband struct to the PID controller.
+ * @param[out] pid The PID structure to configure
+ * @param[in] deadband The external deadband structure to assign
+ */
+void pid_assign_deadband(struct pid *pid, struct pid_deadband *db)
+{
```

Comment:
> Hmm, the pointer assignment stuff is there to be able to disable the deadband between mode switches, while still being able to update it with no fuzz via the settings callback (maybe allow it as option in TxPID)

I don't understand this-- if we have a deadband on the rate controller, shouldn't it always be there-- whether we're flying rate or attitude?

> If we malloc for saving RAM, maybe we should also cause a system configuration error on F1 when any deadbands are enabled the same time as autotune.

We won't survive that long.. so not worth putting in the check that way.  Solution to surviving bad config on F1 is more complicated (and not served by adding special purpose code for each bad combination, unfortunately).